mteb/tasks/Classification/multilingual/SIB200Classification.py

```diff
@@ -0,0 +1,263 @@
+    "tel": ["tel-Telu"],
+    "tgk": ["tgk-Cyrl"],
+    "tgl": ["tgl-Latn"],
+    "tha": ["tha-Thai"],
+    "tir": ["tir-Ethi"],
+    "tpi": ["tpi-Latn"],
+    "tsn": ["tsn-Latn"],
+    "tso": ["tso-Latn"],
+    "tuk": ["tuk-Latn"],
+    "tum": ["tum-Latn"],
+    "tur": ["tur-Latn"],
+    "twi": ["twi-Latn"],
+    "tzm": ["tzm-Tfng"],
+    "uig": ["uig-Arab"],
+    "ukr": ["ukr-Cyrl"],
+    "umb": ["umb-Latn"],
+    "urd": ["urd-Arab"],
+    "uzn": ["uzn-Latn"],
+    "vec": ["vec-Latn"],
+    "vie": ["vie-Latn"],
+    "war": ["war-Latn"],
+    "wol": ["wol-Latn"],
+    "xho": ["xho-Latn"],
+    "ydd": ["ydd-Hebr"],
+    "yor": ["yor-Latn"],
+    "yue": ["yue-Hant"],
+    "zho": ["zho-Hant"],
+    "zsm": ["zsm-Latn"],
+    "zul": ["zul-Latn"],
+}
+
+
+def _transform(dataset):
+    dataset = dataset.class_encode_column("category")
+    dataset = dataset.rename_columns({"category": "label"})
+    dataset = dataset.remove_columns(["index_id"])
+    print(len(dataset["test"]))
+    return dataset
+
+
+class SIB200Classification(MultilingualTask, AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="SIB200Classification",
+        description="""SIB-200 is the largest publicly available topic classification
+        dataset based on Flores-200 covering 205 languages and dialects.
+        """,
+        reference="https://arxiv.org/abs/2309.07445",
+        dataset={
+            "path": "Davlan/sib200",
+            "revision": "38977a667f6fc264d5c26ec57a01e16db040b358",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=_LANGS,
+        main_score="accuracy",
+        date=("2024-01-27", "2024-02-19"),
+        form=["written"],
+        domains=["Constructed"],
+        task_subtypes=["Topic classification"],
+        license="cc-by-sa-4.0",
+        socioeconomic_status="high",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""@article{adelani2023sib,
+            title={SIB-200: A simple, inclusive, and big evaluation dataset for topic classification in 200+ languages and dialects},
+            author={Adelani, David Ifeoluwa and Liu, Hannah and Shen, Xiaoyu and Vassilyev, Nikita and Alabi, Jesujoba O and Mao, Yanke and Gao, Haonan and Lee, Annie En-Shiun},
+            journal={arXiv preprint arXiv:2309.07445},
+            year={2023}
+        }""",
+        n_samples={"test": 204},
+        avg_character_length={"test": 135.53},
+    )
+
+    def load_data(self, **kwargs):
+        """Load dataset from HuggingFace hub"""
+        if self.data_loaded:
+            return
+        self.dataset = {}
+        for lang in self.langs:
+            metadata = self.metadata_dict.get("dataset", None)
+            _lang_code = self.langs[lang][0].replace("-", "_")
+            dataset = datasets.load_dataset(name=_lang_code, **metadata)
+            self.dataset[lang] = _transform(dataset)
```

Comment:
use the dataset transform instead

Reply:
thanks for suggestion! this was inspired by how https://github.com/embeddings-benchmark/mteb/pull/439/files#diff-923acdf97b53930851e1009bea70faab18366725ffdd8a39225ff0ccebe39be4R111 was implemented, so I wanted to remain consistent, but sure, I think I have an idea how to do it via transform.

Reply:
Ahhh, I even reviewed that one - well, it seems like I will have to fix that one

Reply:
Nw, since I'm on it, happy to take that as a part of this PR. The change should be analogous.